src/composables/useCvss4Calculator.ts

```diff
@@ -0,0 +1,90 @@
+import { ref, watch, computed } from 'vue';
+
+import { map } from 'ramda';
+
+import { useFlaw } from '@/composables/useFlaw';
+
+import { IssuerEnum } from '@/generated-client';
+import { CVSS40, METRICS } from '@/utils/cvss40';
+import { isNonArrayObject } from '@/utils/helpers';
+import type { Dict } from '@/types';
+import { CvssVersions } from '@/constants';
+
+function rhCvss4_0() {
+  const { flaw } = useFlaw();
+
+  return flaw.value.cvss_scores.find(
+    cvss => cvss.issuer === IssuerEnum.Rh && cvss.cvss_version === CvssVersions.V4,
+  );
+}
+
+const deepMap = (transform: (arg: any) => any, object: Record<string, any>): any =>
+  map(
+    (val: any) => isNonArrayObject(val)
+      ? deepMap(transform, val)
+      : transform(val)
+    , object,
+  );
+
+export function useCvss4Calculator() {
+  const { flaw } = useFlaw();
+
+  const cvss4Selections = ref(
+    deepMap(value => Array.isArray(value) ? null : value, METRICS),
+  );
+  const cvss4Vector = computed(() => {
+    const vector = Object.values(cvss4Selections.value).flatMap((group: any) =>
+      Object.entries(group)
+        .filter(([, value]) => value !== null)
+        .map(([metric, value]) => `${metric}:${value}`),
+    ).join('/');
+    return `CVSS:4.0/${vector}`;
```

Comment:
This is setting the string `CVSS:4.0/` always, even if the vector is empty, this seems to introduce a werid behavior in the component, as jumping from an empty CvssV3 to a V4, now suddenly the vector is invalid, and since is the default behavior, there's no way of clearing the input, thankfully the V3 can be set to 0, if not the user would be stuck with an invalid flaw until a random vector is typed or the tab reloaded.

Also for some reason the V4 calculator sets an score of 1 when the input is empty




https://github.com/user-attachments/assets/6df93d6a-8b76-469f-b880-cc8ca5cd5ef4